### PR TITLE
Add Model Tests

### PR DESCRIPTION
**Pull request by javier-aguilar**
- Add model test for order#items_by_merchant
- Add model test for item#fulfillable?(num)
- Add model test for item#item_order(order_id)
- Add model test for item#fulfilled_inventory(order_id)
- 100% coverage for spec/models

### CODE REVIEW

File: spec/models/item_spec.rb [hunk 1]
```diff
@@ -130,5 +130,29 @@
       order.item_orders.create(item: @chain, price: @chain.price, quantity: 2)
       expect(@chain.no_orders?).to eq(false)
     end
+
+    it 'item#fulfillable?(num)' do
```
Comment (mlynch5187) [inline spec/models/item_spec.rb:134]: Love what you are doing here! This really makes it easy to know what's being tested.

**Review by mlynch5187 — APPROVED** [review]
I really like the way you organized these tests! Would it make sense to model our feature tests this way?

File: spec/models/item_spec.rb (hunk 1 above)
Comment (reid-andrew) [reply spec/models/item_spec.rb:134]: Agreed. I'd be good with applying this naming convention across all model tests.

**Review by reid-andrew — APPROVED** [review]

File: spec/models/order_spec.rb [hunk 2]
```diff
@@ -97,5 +97,10 @@
       expect(@order_1.total_value_by_merchant(@brian)).to eq(30)
     end
 
+    it 'order#items_by_merchant(merchant_id)' do
+      expect(@order_1.items_by_merchant(@meg.id)).to eq([@tire])
+      expect(@order_1.items_by_merchant(@meg.id)).not_to eq([@tire, @pull_toy])
+    end
+
```
Comment (aperezsantos) [inline spec/models/order_spec.rb:104]: Just need to remove extra space.

**Review by aperezsantos — APPROVED** [review]
This all looks great! Thank you.

**Pull request merged**